pkg/leeway/sbom.go

```diff
@@ -96,6 +102,142 @@ func GetSBOMParallelism(sbomConfig WorkspaceSBOM) int {
+func generateDeterministicUUID(content []byte) string {
+	// Use UUIDv5 (SHA-1 based) with the standard DNS namespace UUID.
+	// The DNS namespace (6ba7b810-9dad-11d1-80b4-00c04fd430c8) is defined in RFC 4122
+	// and commonly used for generating deterministic UUIDs from content.
+	namespace := uuid.MustParse("6ba7b810-9dad-11d1-80b4-00c04fd430c8")
+	
+	return uuid.NewSHA1(namespace, content).String()
+}
+
+// normalizeCycloneDX makes CycloneDX SBOM deterministic
+func normalizeCycloneDX(sbomPath string, timestamp time.Time) error {
+	data, err := os.ReadFile(sbomPath)
+	if err != nil {
+		return fmt.Errorf("failed to read SBOM: %w", err)
+	}
+
+	var sbom map[string]interface{}
+	if err := json.Unmarshal(data, &sbom); err != nil {
+		return fmt.Errorf("failed to parse SBOM: %w", err)
+	}
+
+	// Normalize timestamp
+	metadata, ok := sbom["metadata"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("metadata field not found or invalid")
+	}
+	metadata["timestamp"] = timestamp.Format(time.RFC3339)
+
+	// Generate deterministic UUID from normalized content (without timestamp and UUID)
+	// Remove non-deterministic fields before hashing
+	delete(sbom, "serialNumber")
+	normalizedForHash, err := json.Marshal(sbom)
+	if err != nil {
+		return fmt.Errorf("failed to marshal SBOM for hashing: %w", err)
+	}
+	contentHash := sha256.Sum256(normalizedForHash)
+	deterministicUUID := generateDeterministicUUID(contentHash[:])
+	sbom["serialNumber"] = fmt.Sprintf("urn:uuid:%s", deterministicUUID)
+
+	// Write back
+	normalized, err := json.MarshalIndent(sbom, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal SBOM: %w", err)
+	}
+
+	return os.WriteFile(sbomPath, normalized, 0644)
+}
+
+// normalizeSPDX makes SPDX SBOM deterministic
+func normalizeSPDX(sbomPath string, timestamp time.Time) error {
+	data, err := os.ReadFile(sbomPath)
+	if err != nil {
+		return fmt.Errorf("failed to read SBOM: %w", err)
+	}
+
+	var sbom map[string]interface{}
+	if err := json.Unmarshal(data, &sbom); err != nil {
+		return fmt.Errorf("failed to parse SBOM: %w", err)
+	}
+
+	// Normalize timestamp
+	creationInfo, ok := sbom["creationInfo"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("creationInfo field not found or invalid")
+	}
+	creationInfo["created"] = timestamp.Format(time.RFC3339)
+
+	// Generate deterministic UUID from normalized content (without timestamp and UUID)
+	// Save original namespace to extract later
+	originalNamespace, _ := sbom["documentNamespace"].(string)
+	delete(sbom, "documentNamespace")
+	
+	normalizedForHash, err := json.Marshal(sbom)
+	if err != nil {
+		return fmt.Errorf("failed to marshal SBOM for hashing: %w", err)
+	}
+	contentHash := sha256.Sum256(normalizedForHash)
+	deterministicUUID := generateDeterministicUUID(contentHash[:])
+
+	// Replace UUID in documentNamespace
+	if originalNamespace != "" {
+		// Extract base URL and replace UUID at the end
+		lastDash := strings.LastIndex(originalNamespace, "-")
```

Comment:
Do you think it would make sense to validate the format of `originalNamespace` here?

- Assumes UUID is at the end of the namespace
- Doesn't validate that what it's replacing is actually a UUID
- Could fail silently if the namespace format is unexpected
- No test coverage for edge cases (namespace without UUID, UUID in middle, etc.)

Something like this:

```
uuidPattern := regexp.MustCompile(`[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}`)
if uuidPattern.MatchString(originalNamespace) {
    originalNamespace = uuidPattern.ReplaceAllString(originalNamespace, deterministicUUID)
}
```